React/Modules/RCTUIManager.m

```diff
@@ -79,13 +81,30 @@ static UIViewAnimationOptions UIViewAnimationOptionsFromRCTAnimationType(RCTAnim
       return UIViewAnimationOptionCurveEaseInOut;
     case RCTAnimationTypeKeyboard:
       // http://stackoverflow.com/questions/18870447/how-to-use-the-default-ios7-uianimation-curve
-      return (UIViewAnimationOptions)(7 << 16);
+      return (UIViewAnimationOptions)(_currentKeyboardAnimationCurve << 16);
     default:
       RCTLogError(@"Unsupported animation type %zd", type);
       return UIViewAnimationOptionCurveEaseInOut;
   }
 }
 
++ (void)initializeStatics
```

Comment:
I can't use `+initialize` here. I'll add a comment describing why. Here's the reasoning: `+initialize` runs immediately before the first message is sent to the class which may be too late for us. By this time, we may have missed some `UIKeyboardWillChangeFrameNotification`s.

I might be able to use `+load` but there are some [caveats when using `+load`](https://www.mikeash.com/pyblog/friday-qa-2009-05-22-objective-c-class-loading-and-initialization.html) because it is executed so early so I feel more comfortable with a custom initialization function.